spec/features/consumer/caching/shops_caching_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+feature "Shops caching", js: true, caching: true do
+  let!(:distributor) { create(:distributor_enterprise, with_payment_and_shipping: true, is_primary_producer: true) }
+  let!(:order_cycle) { create(:open_order_cycle, distributors: [distributor], coordinator: distributor) }
+
+  describe "caching enterprises AMS data" do
+    it "caches data for all enterprises, with the provided options" do
+      visit shops_path
+
+      key, options = CacheService::FragmentCaching.ams_shops
+      expect_cached "views/#{key}", options
+    end
+
+    it "keeps data cached for a short time on subsequent requests" do
+      # One minute ago...
+      Timecop.travel(Time.zone.now - 1.minute) do
+        visit shops_path
+
+        expect(page).to have_content distributor.name
+
+        distributor.name = "New Name"
+        distributor.save
```

Comment:
Always good to use `save!` in specs. If `save` fails this spec would always pass.

Reply:
Done.